CMakeLists.txt

```diff
@@ -99,7 +99,8 @@ set(xgboost_exports "")
 set(xgboost_targets "")
 
 if(USE_CUDA)
-  
+
+  set(CUDA_LINK_LIBRARIES_KEYWORD PUBLIC)
```

Comment:
Not used?

Reply:
... in `FindCUDA.cmake`
https://github.com/Kitware/CMake/blob/e40541339ae8a780bc410f0ef8b22d01e594c1ba/Modules/FindCUDA.cmake#L85-L90

else it uses the plain `target_link_libraries()` and we are forced to use that in the parent project for the same lib.

Reply:
And ... https://github.com/Kitware/CMake/blob/e40541339ae8a780bc410f0ef8b22d01e594c1ba/Modules/FindCUDA.cmake#L1822-L1827

Reply:
Okay, I see